controllers/clustergroupupgrade_controller.go

```diff
@@ -973,34 +1141,148 @@ func (r *ClusterGroupUpgradeReconciler) reconcileResources(ctx context.Context,
 	return nil
 }
 
-func (r *ClusterGroupUpgradeReconciler) buildRemediationPlan(ctx context.Context, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
+func (r *ClusterGroupUpgradeReconciler) isClusterNonCompliantWithPolicy(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, clusterName string, policy *unstructured.Unstructured) bool {
+	// Get the status part of the policy.
+	statusObject := policy.Object["status"].(map[string]interface{})
+	// Get the part of the status that holds the clusters' compliance.
+	statusCompliance := statusObject["status"]
+
+	// If all the clusters are compliant, return nil.
+	if statusCompliance == nil {
+		return false
+	}
+
+	subStatus := statusCompliance.([]interface{})
+
+	// Loop through all the NonCompliant clusters in the policy's status.
+	for _, crtSubStatusCrt := range subStatus {
+		crtSubStatusMap := crtSubStatusCrt.(map[string]interface{})
+		// If the cluster is NonCompliant, add it to the list of NonCompliant clusters.
```

Comment:
The comment says it should be added to the list but that is not done here.

Reply:
Oops, missed it when I restructured this function. Thank you.